raft/src/main/java/org/apache/kafka/raft/QuorumState.java

```diff
@@ -62,8 +62,8 @@ public void initialize(OffsetAndEpoch logEndOffsetAndEpoch) throws IOException {
             log.warn("Clear local quorum state store {}", store.toString(), e);
             store.clear();
 
-            election = ElectionState.withUnknownLeader(0);
-            state = new FollowerState(election.epoch);
+            election = ElectionState.withUnknownLeader(0, voters);
```

Comment:
In the case that the state is successfully loaded from the file, should we validate that it matches the configured set of voters?

Reply:
Good question, I feel we should fail the initialization for the first version as the state file takes precedence and user should always ensure to update the configuration for all or nothing.